### ISSUE TITLE
[Question] Code stuck in epoch 0

### ISSUE THREAD

**Issue by shubashri123** [delira-dev/delira#157]
When I try to run your original code with your dataset itself,  its not exceeding epoch_0 and no values come for that single epoch.
This is my thing 

WARNING:delira.training.pytorch_trainer:Apex was not found found, trying to                                     continue in full precision instead
Epoch 1:   0%|          | 0/157 [00:00<?, ? batch/s]

I made run it for 2 days but no update. please help me out.
 @justusschock
sorry for not given those information.
  I run that in colab. let me share you the notebook.
thanks

**Comment by justusschock** [comment]
We cannot help you without any information regarding your system, which code you ran, which `delira` version you're using etc.  

The above warning is not crucial, but indicates, that you tried running in mixed precision without installing Apex, which should work nevertheless (and it does for me)

So, please provide the following information:

* What OS are you on?
* What is your machine configuration?
* How did you install `delira`?
* Which `delira` version are you using?
* Which python version are you using?
* Are you using an environment management system (i.e. `conda` or `virtualenv`)?
* Are you using a fresh python install?
and finally:
Which code did you try to run?

**Comment by shubashri123** [comment]
i run segmentation_3d_pytorch.ipynb. Exactly the same code i have run and I didn't do even a single change. I made run it in colab. 

the code is 

!pip install imgaug==0.2.5 > /dev/null
!pip install imageio > /dev/null
!pip install delira > /dev/null

logger = None
import torch
from delira.training import Parameters
params = Parameters(fixed_params={
    "model": {
        "in_channels": 1, 
        "num_classes": 4
    },
    "training": {
        "batch_size": 64, # batchsize to use
        "num_epochs": 10, # number of epochs to train
        "optimizer_cls": torch.optim.Adam, # optimization algorithm to use
        "optimizer_params": {'lr': 1e-3}, # initialization parameters for this algorithm
        "losses": {"CE": torch.nn.CrossEntropyLoss()}, # the loss function
        "lr_sched_cls": None,  # the learning rate scheduling algorithm to use
        "lr_sched_params": {}, # the corresponding initialization parameters
        "metrics": {} # and some evaluation metrics
    }
})

  from io import BytesIO
  from zipfile import ZipFile
  from urllib.request import urlopen

  resp = urlopen("http://www.fmrib.ox.ac.uk/primers/intro_primer/ExBox3/ExBox3.zip")
  zipfile = ZipFile(BytesIO(resp.read()))
  zipfile_list = zipfile.namelist()
  print(zipfile_list)
  img_file = zipfile.extract("ExBox3/T1_brain.nii.gz")
  mask_file = zipfile.extract("ExBox3/T1_brain_seg.nii.gz")
 

import SimpleITK as sitk
import numpy as np

# load image and mask
img = sitk.GetArrayFromImage(sitk.ReadImage(img_file))
img = img.astype(np.float32)
mask = mask = sitk.GetArrayFromImage(sitk.ReadImage(mask_file))
mask = mask.astype(np.float32)

assert mask.shape == img.shape
print(img.shape)

from delira.data_loading import AbstractDataset

class CustomDataset(AbstractDataset):
    def __init__(self, img, mask, num_samples=1000):
        super().__init__(None, None)
        self.data = {"data": img.reshape(1, *img.shape), "label": mask.reshape(1, *mask.shape)}
        self.num_samples = num_samples

    def __getitem__(self, index):
        return self.data

    def __len__(self):
        return self.num_samples

dataset_train = CustomDataset(img, mask, num_samples=10000)

dataset_val = CustomDataset(img, mask, num_samples=1)

from batchgenerators.transforms import ContrastAugmentationTransform, Compose
from batchgenerators.transforms.sample_normalization_transforms import MeanStdNormalizationTransform

transforms = Compose([
    ContrastAugmentationTransform(), # randomly adjust contrast
    MeanStdNormalizationTransform(mean=[img.mean()], std=[img.std()])]) # use concrete values since we only have one sample (have to estimate it over whole dataset otherwise)

from delira.data_loading import BaseDataManager, SequentialSampler, RandomSampler

manager_train = BaseDataManager(dataset_train, params.nested_get("batch_size"),
                                transforms=transforms,
                                sampler_cls=RandomSampler,
                                n_process_augmentation=4)

 manager_val = BaseDataManager(dataset_val, params.nested_get("batch_size"),
                               transforms=transforms,
                               sampler_cls=SequentialSampler,
                               n_process_augmentation=4)

import warnings
warnings.simplefilter("ignore", UserWarning) # ignore UserWarnings raised by dependency code
warnings.simplefilter("ignore", FutureWarning) # ignore FutureWarnings raised by dependency code


from delira.training import PyTorchExperiment
from delira.training.train_utils import create_optims_default_pytorch
from delira.models.segmentation import UNet3dPyTorch

if logger:
    logger.info("Init Experiment")
experiment = PyTorchExperiment(params, UNet3dPyTorch,
                               name="Segmentation3dExample",
                               save_path="./tmp/delira_Experiments",
                               optim_builder=create_optims_default_pytorch,
                               gpu_ids=[0], mixed_precision=True)
experiment.save()

model = experiment.run(manager_train, manager_val)
